### PR TITLE
Kernel intent hash

### LINKED ISSUE #119: Implement the KERNEL_INTENT_TYPEHASH constant and hashKernelIntent function in ConsensusGatewayBase contract
- Create a constant `bytes32` variable `KERNEL_INTENT_TYPEHASH` for storing kernel intent type hash. 
  - This will be used in `hashKernelIntent` function for the creation of `kernelIntentHash_`
```solidity
bytes32 public constant KERNEL_INTENT_TYPEHASH = keccak256(
    abi.encode(
        "KernelIntent(uint256 height,bytes32 kernelHash)"
    )
)
```
- Create the function `hashKernelIntent` 
  - Parameters:
    - `uint256 _height` and
    - `bytes32 _kernelHash`
  - Returns:
    - `bytes32 kernelIntentHash_`
  - This function will be called from `declareOpenKernel` function 
```solidity
function hashKernelIntent(
    uint256 _height,
    bytes32 _kernelHash
) 
    public 
    returns (bytes32 kernelIntentHash_) 
```
- The function `hashKernelIntent` will be used by `ConsensusGateway` and `ConsensusCogateway` contracts

### PR DESCRIPTION
**Pull request by 0xsarvesh**
fixes #119 

Added method to hash kernel intent.

### CODE REVIEW

File: contracts/consensus-gateway/ConsensusGatewayBase.sol [hunk 1]
```diff
@@ -19,6 +19,14 @@ import "../ERC20I.sol";
 /** Base contract for ConsensusGateway and ConsensusCogateway contracts. */
 contract ConsensusGatewayBase {
 
+    /* Constants */
```
Comment (gulshanvasnani) [inline contracts/consensus-gateway/ConsensusGatewayBase.sol:22]: Add newline.

File: test/consensus-gateway/hash_kernel_intent.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+const BN = require('bn.js');
+
+const ConsensusGatewayBase = artifacts.require('ConsensusGatewayBaseTest');
+
+contract('ConsensusGatewayBase::hashKernelIntent', (accounts) => {
+  it('should hash kernel content', async () => {
+    const consensusGatewayBase = await ConsensusGatewayBase.new();
+
+    const height = new BN(1);
```
Comment (gulshanvasnani) [inline test/consensus-gateway/hash_kernel_intent.js:25]:
```suggestion
    const height = new BN(Utils.getRandomNumber(500));
```

File: test/consensus-gateway/hash_kernel_intent.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    const kernelHash = '0xb72db4929fec6eb0a73772f5eb48498cceff24c8449d805d5ddbffd87838a9d8';
+
+    const kernelIntentHash = await consensusGatewayBase.hashKernelIntent.call(height, kernelHash);
+
+    const expectedIntentHash = '0x942ad9797599fd9964cee5b758766914683f3b84a483371612861e27eadd125e';
```
Comment (gulshanvasnani) [inline test/consensus-gateway/hash_kernel_intent.js:30]: Instead of hardcoding of **expectedIntentHash**, calculated it.

**Review by gulshanvasnani — COMMENTED** [review]
Couple of inline comment.

File: test/consensus-gateway/hash_kernel_intent.js (hunk 2 above)
Comment (0xsarvesh) [reply test/consensus-gateway/hash_kernel_intent.js:25]: Created discussion point can be discussed with the team. 
https://hackmd.io/1rni8bzWQo2_wnqrpDn59Q

**Review by gulshanvasnani — APPROVED** [review]
LGTM 🚀

**Review by deepesh-kn — APPROVED** [review]
LGTM 👍

**Pull request merged**